src/ray/object_manager/object_manager.cc

```diff
@@ -119,127 +122,41 @@ ray::Status ObjectManager::Pull(const ObjectID &object_id) {
     RAY_LOG(ERROR) << object_id << " attempted to pull an object that's already local.";
     return ray::Status::OK();
   }
-  if (pull_requests_.find(object_id) != pull_requests_.end()) {
+
+  bool subscribe_to_locations;
+  bool start_timer;
+  pull_manager_.PullObject(object_id, &subscribe_to_locations, &start_timer);
+  if (start_timer) {
+    RAY_CHECK(pull_timers_.find(object_id) == pull_timers_.end());
```

Comment:
Is it possible to put the `pull_timers_` and related code into `PullManager`?

Reply:
I was tempted to do so but looking back at Robert's pr (#3826) and [Stephanie's comment](https://github.com/ray-project/ray/pull/3826#pullrequestreview-196665125), it is a lot cleaner to keep the `pull_timers_` outside of the `PullManager`. However, leaving the two separate does bring up a lot of subtle edge cases.

@robertnishihara do you have any thoughts you want to add?